File: Java/AlgoExpert/Searching/Easy/Find Three Largest Numbers.java
import java.util.*;

class Program {
  public static int[] findThreeLargestNumbers(int[] array) {
    // Write your code here.
		int[] largestNums = {Integer.MIN_VALUE, Integer.MIN_VALUE, Integer.MIN_VALUE};
		
		for(int i = 0; i < array.length; i++){
			if(array[i] > largestNums[2]){
				largestNums[0] = largestNums[1];
				largestNums[1] = largestNums[2];
				largestNums[2] = array[i];
			}else if(array[i] > largestNums[1]){
				largestNums[0] = largestNums[1];
				largestNums[1] = array[i];
			}else if(array[i] > largestNums[0]){
				largestNums[0] = array[i];
			}
		}
		
    return largestNums;
  }
}

File: Python/AlgoExpert/Searching/Easy/findThreeLargestNumbers.py
def findThreeLargestNumbers(array):
    # Write your code here.
    first, second, third = float('-inf'), float('-inf'), float('-inf')

    for num in array:
        if num >= first:
            third = second
            second = first
            first = num
        elif num >= second:
            third = second
            second = num
        elif num >= third:
            third = num

    return [third, second, first]
